proposals/csharp-7.2/readonly-ref.md

```diff
@@ -40,105 +58,160 @@ static Vector3 Add (ref readonly Vector3 v1, ref readonly Vector3 v2)
     foo(ref v1.X);
 
     // OK
-    return new Vector3(v1.X +v2.X, v1.Y + v2.Y, v1.Z + v2.Z);
+    return new Vector3(v1.X + v2.X, v1.Y + v2.Y, v1.Z + v2.Z);
 }
 ```
 
-## Syntax
+`in` cannot be combined with other parameter passing modifiers such as `out`, `ref`, `params`. 
 
-One of proposed syntaxes is to use existing `in` keyword as a shorter form of `ref readonly`. It could be that both syntaxes are allowed or that we pick just one of them.
+## Use of `in` in signatures.
 
-While syntax is TBD, I will use `in` as it is significantly shorter than `ref readonly`.
+- `in` parameters and returns are allowed anywhere where ordinary byval parameters are allowed. This includes indexers, operators (including conversions), delegates, lambdas, local functions.
```

Comment:
> in paramaters and returns

There is no `in` return. May just need to omit `returns` here as they will be covered in the `ref readonly` return section.

Reply:
right. this is a copy/paste from old doc. 
returns are definitely not allowed in place of parameters :-)

---
In reply to: [154465814](https://github.com/dotnet/csharplang/pull/1165#discussion_r154465814) [](ancestors = 154465814)